hisensetv/__init__.py

```diff
@@ -218,62 +218,93 @@ def _send_key(self, keyname: str):
         Args:
             keyname: Name of the key press to send.
         """
-        self._call_service(service="remote_service", action="sendkey", payload=keyname)
+        try:
+           self._call_service(service="remote_service", action="sendkey", payload=keyname)
+           result = True
+        except json.JSONDecodeError:
```

Comment:
I am a little confused by this, what is raising the `JSONDecodeError`?  Is this something that can be fixed elsewhere?

Reply:
Hmm, well. I have absolutely no idea where that came from... lol.
That line should just be "except:"

Reply:
What exception is getting raised there?  It would be better to scope it down a little, a bare `except` will catch system level events like `KeyboardInterrupt` and `SystemExit`.

Reply:
I'd have to defer to you here, as my best practices knowledge on exception handling is limited.

Reply:
Generally I try to propagate exceptions as much as possible to the user - unless the exceptions are errors due to the internal code rather than a device malfunction or user error.

If this is an internal code error it would be helpful to know what causes it so that we could perhaps change the internal workings to function better, and limit the scope to capture just the exception(s) that are problematic.  Are there any specific exceptions that you have had?  Do you have the traceback for them?

Reply:
No, I experienced no exceptions in my development. My thought process was to always return a bool value, based on success of send_key command. Any exception failure would be captured and appropriate return value sent.